lldb/include/lldb/Target/SyntheticFrameProvider.h

```diff
@@ -0,0 +1,133 @@
+
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TARGET_SYNTHETICFRAMEPROVIDER_H
+#define LLDB_TARGET_SYNTHETICFRAMEPROVIDER_H
+
+#include "lldb/Core/PluginInterface.h"
+#include "lldb/Target/StackFrameList.h"
+#include "lldb/Target/ThreadSpec.h"
+#include "lldb/Utility/ScriptedMetadata.h"
+#include "lldb/Utility/Status.h"
+#include "lldb/lldb-forward.h"
+#include "llvm/Support/Error.h"
+
+#include <optional>
+#include <vector>
+
+namespace lldb_private {
+
+/// Descriptor for configuring a synthetic frame provider.
+///
+/// This struct contains the metadata needed to instantiate a frame provider
+/// and optional filters to control which threads it applies to.
```

Comment:
This is also pretty vague and seems very "Scripted*" specific. If I'm implementing a C++ based SyntheticFrameProviderDescriptor (e.g. for Python frames), wy do I need to provided scripted metadata? From a design POV, I would expect the constructor to take the stuff that's encapsulated here, and this is part of the "Scripted*" bridging.

Reply:
Ok, I made a different callback for "native" frame providers that doesn't require SyntheticFrameProviderDescriptor. However I still need this in lldb core since the targets needs to hold them, in case it doesn't have a process (and threads) yet. I left it here but I could move it to Utility next to ScriptedMetadata if you prefer.